addons/mail/models/mail_message.py

```diff
@@ -1171,11 +1171,17 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
-        updates = [[
-            (self._cr.dbname, 'res.partner', author.id),
-            {'type': 'message_notification_update', 'elements': self.env['mail.message'].concat(*author_messages)._message_notification_format()}
-        ] for author, author_messages in groupby(messages.sorted('author_id'), itemgetter('author_id'))]
-        self.env['bus.bus'].sendmany(updates)
+        notifications = []
+        partner_ids = list()
+        for user in self.env['res.users.log'].search([]):
+            partner = self.env['res.users'].search([('id', '=', int(user.create_uid))])
+            partner_ids.append(partner.partner_id.id)
+        for author, author_messages in groupby(messages.sorted('author_id'), itemgetter('author_id')):
+            partner_ids.append(author.id)
+            data = {'type': 'message_notification_update', 'elements': self.env['mail.message'].concat(*author_messages)._message_notification_format()}
+            for partner in list(set(partner_ids)):
+                notifications.append([(self._cr.dbname, 'res.partner', partner), data])
+        self.env['bus.bus'].sendmany(notifications)
```

Comment:
I think this can be changed to be a bit more readable.

So what we want is a set of partners, and for each partner a set of messages to notify.

The easiest (and most efficient) data structure to store this is a dict of set.
Each key of the dict is a partner, and each value of the dict is the set of messages to notify.

Then you can remove the groupby and other stuff, and just have 1 line to insert the current partner into the dict (with all messages), then 1 loop on all messages to add their author into the dict, and then add the respective message into the set for this author.

And finally one simple loop on the resulting dict to convert into format expected by `sendmany`.